src/functionality.js

```diff
@@ -0,0 +1,124 @@
+// eslint-disable-next-line import/no-cycle
+import interaction from './interactive.js';
+
+export default (e, textField, todoListArray) => {
+  if (textField.value === '') return;
+  e.preventDefault();
+  const localStorageObject = {
+    description: textField.value,
+    completed: false,
+    index: todoListArray.length + 1,
+  };
+  todoListArray.push(localStorageObject);
+  localStorage.setItem('listOfTasks', JSON.stringify(todoListArray));
+};
+
+const editTasks = (array) => {
+  const individualTasks = document.querySelectorAll('.task-description');
+  individualTasks.forEach((tasks, taskIndex) => {
+    tasks.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        array.filter((object, objectIndex) => {
+          if (taskIndex === objectIndex) {
+            object.description = tasks.value;
+            tasks.style.outline = 'none';
+            tasks.blur();
+          }
+          return false;
+        });
+      }
+      localStorage.setItem('listOfTasks', JSON.stringify(array));
+    });
+  });
+};
+
+export const clickEditButton = () => {
+  const editHandler = document.querySelectorAll('.edit');
+  editHandler.forEach((edit) => {
+    const textDescription = document.getElementById(edit.dataset.target);
+    edit.addEventListener('click', () => {
+      textDescription.focus();
+      const moreButton = document.querySelectorAll('.more');
+      moreButton.forEach((btn) => {
+        const dropdownMenu = document.getElementById(btn.dataset.target);
+        dropdownMenu.classList.remove('show');
+      });
+    });
+  });
+};
+
+export const clickDeleteButton = (array) => {
+  const deleteHandler = document.querySelectorAll('.delete');
+  deleteHandler.forEach((handler, index) => {
+    handler.addEventListener('click', () => {
+      const filteredArray = array.filter((task) => array.indexOf(task) !== index);
+      filteredArray.forEach((obj, index) => {
+        obj.index = index + 1;
+      });
+      localStorage.setItem('listOfTasks', JSON.stringify(filteredArray));
+      window.location.reload();
+    });
+  });
+};
+
+export const deleteCompletedTask = (array) => {
+  const deleteCompletedBtn = document.querySelector('.clear-div');
+  deleteCompletedBtn.addEventListener('click', () => {
+    const filteredArray = array.filter((obj) => obj.completed !== true);
+    filteredArray.forEach((obj, index) => {
+      obj.index = index + 1;
+    });
+    localStorage.setItem('listOfTasks', JSON.stringify(filteredArray));
+    window.location.reload();
+  });
+};
+
+export const restartTask = () => {
+  const reloadButton = document.querySelector('.reload');
+  reloadButton.addEventListener('click', () => {
+    localStorage.setItem('listOfTasks', JSON.stringify([]));
+    window.location.reload();
```

Comment:
The codes for setting local storage is being repeated multiple times, It could be better if you wrap them inside one function and call it wherever you want 
```suggestion
    localStorage.setItem('listOfTasks', JSON.stringify([]));
    window.location.reload();
```

Reply:
Thank you for the feedback, I have implemented the changes.